From 5a02b2b34587ef62b7493ff8cd2d726a407dc98c Mon Sep 17 00:00:00 2001
From: Leo Wang <leo.jt.wang@fii-foxconn.com>
Date: Wed, 28 May 2025 21:01:06 +0800
Subject: [PATCH 1002/1003] ARM dts aspeed clemente add various device nodes
 and properties

This commit updates the device tree for the Facebook Clemente BMC
to include configurations for new and existing hardware components.

The key changes include:
- Corrected the compatible string for UART3 to 'aspeed,ast2600-uart'
  to ensure proper driver binding.
- Enhanced MAC2 and MAC3 nodes with NCSI-specific properties
  'ncsi-ctrl,start-redo-probe' and 'ncsi-package', enabling
  advanced NCSI functionalities for network controllers.
- Configured fan controller nodes (MAX31790) with channel-specific
  TACH sensor types and addressing properties, improving fan speed
  monitoring capabilities.
- Updated I2C7 bus settings with a faster frequency and multi-master
  mode, and introduced an IPMB device node for USB Debug Connector,
  facilitating BMC communication.
- Added alert GPIO and timeout settings for the SSIF BMC device,
  improving the robustness of BMC communication.
- Integrated several TI INA230 power sensors on I2C12 mux channel 7,
  including their shunt resistor values, to enable comprehensive
  power consumption monitoring across the board.

These changes ensure the BMC's device tree accurately reflects the
hardware configuration and enables necessary driver support for
these components.

Signed-off-by: Leo Wang <leo.jt.wang@fii-foxconn.com>
Upstream-Status: Unknown
---
 .../aspeed/aspeed-bmc-facebook-clemente.dts   | 58 +++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts
index 63b5a13214f1..a702f8641958 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-clemente.dts
@@ -163,6 +163,7 @@ &uart1 {
 };
 
 &uart3 {
+	compatible = "aspeed,ast2600-uart";
 	status = "okay";
 };
 
@@ -179,6 +180,8 @@ &mac2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_ncsi3_default>;
 	use-ncsi;
+	ncsi-ctrl,start-redo-probe;
+	ncsi-package = <1>;
 };
 
 &mac3 {
@@ -186,6 +189,8 @@ &mac3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_ncsi4_default>;
 	use-ncsi;
+	ncsi-ctrl,start-redo-probe;
+	ncsi-package = <2>;
 };
 
 &fmc {
@@ -505,10 +510,30 @@ i2c1mux0ch2: i2c@2 {
 			fanctl0: fan-controller@21{
 				compatible = "maxim,max31790";
 				reg = <0x21>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				channel@2 {
+					reg = <2>;
+					sensor-type = "TACH";
+				};
+				channel@5 {
+					reg = <5>;
+					sensor-type = "TACH";
+				};
 			};
 			fanctl1: fan-controller@27{
 				compatible = "maxim,max31790";
 				reg = <0x27>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				channel@2 {
+					reg = <2>;
+					sensor-type = "TACH";
+				};
+				channel@5 {
+					reg = <5>;
+					sensor-type = "TACH";
+				};
 			};
 		};
 		i2c1mux0ch3: i2c@3 {
@@ -663,6 +688,16 @@ eeprom@51 {
 
 &i2c7 {
 	status = "okay";
+	bus-frequency = <100000>;
+	multi-master;
+	aspeed,hw-timeout-ms = <1000>;
+
+	//USB Debug Connector
+	ipmb@10 {
+		compatible = "ipmb-dev";
+		reg = <(0x10 | I2C_OWN_SLAVE_ADDRESS)>;
+		i2c-protocol;
+	};
 };
 
 &i2c8 {
@@ -721,6 +756,8 @@ &i2c11 {
 	ssif-bmc@10 {
 		compatible = "ssif-bmc";
 		reg = <0x10>;
+		alert-gpios = <&gpio0 ASPEED_GPIO(C, 7) GPIO_ACTIVE_LOW>;
+		timeout-ms = <5000>;
 	};
 };
 
@@ -779,6 +816,27 @@ i2c12mux0ch7: i2c@7 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <7>;
+
+			power-sensor@40 {
+				compatible = "ti,ina230";
+				reg = <0x40>;
+				shunt-resistor = <2000>;
+			};
+			power-sensor@41 {
+				compatible = "ti,ina230";
+				reg = <0x41>;
+				shunt-resistor = <2000>;
+			};
+			power-sensor@44 {
+				compatible = "ti,ina230";
+				reg = <0x44>;
+				shunt-resistor = <2000>;
+			};
+			power-sensor@49 {
+				compatible = "ti,ina230";
+				reg = <0x49>;
+				shunt-resistor = <2000>;
+			};
 		};
 	};
 };
-- 
2.49.0

